src/main/java/it/gov/pagopa/pu/migration/wf/activity/ingestion/BaseMigrationFileTypeHandlerActivity.java

```diff
@@ -141,22 +141,26 @@ protected MigrationFileResult handleFilesUpload(
     if (retrievedFiles == null || retrievedFiles.isEmpty()) {
       throw new InvalidMigrationFileException("No file found in the uploaded archive");
     }
-    Organization organization = organizationSearchClient.getByOrganizationId(
-      upload.getOrganizationId(),
-      authnService.getAccessToken()
-    );
+
     List<IngestionFlowFile> filesUploaded = new ArrayList<>(retrievedFiles.size());
     for (Path file : retrievedFiles) {
       String fileName = file.getFileName().toString();
+      String ipaCodeFile = extractIpaCodeFromFileName(fileName);
+
+      Organization organization = organizationSearchClient.getByIpaCode(ipaCodeFile, authnService.getAccessToken());
```

Comment:
let's change dependency to avoid use directly client, let's use OrganizationService instead (where we could configure some caching behavior)